mat/matrix.go

```diff
@@ -207,6 +207,72 @@ type ColNonZeroDoer interface {
 	DoColNonZero(j int, fn func(i, j int, v float64))
 }
 
+// untranspose untransposes a matrix if applicable. If a is an Untransposer, then
+// untranspose returns the underlying matrix and true. If it is not, then it returns
+// the input matrix and false.
+func untranspose(a Matrix) (Matrix, bool) {
+	if ut, ok := a.(Untransposer); ok {
+		return ut.Untranspose(), true
+	}
+	return a, false
+}
+
+// untransposeExtract first untransposes the matrix. It then returns the
```

Comment:
```
// untransposeExtract returns an untransposed matrix in a built-in matrix type.
// 
// The untransposed matrix is return unaltered if it is a built-in matrix type. Otherwise,
// if it implements a Raw method, an appropriate built-in type value is return holding 
// the raw matrix value of the input. If neither of these is possible, the untransposed
// matrix is returned.
```

Reply:
Thanks. Much better.